In [1]:
import pandas as pd
whoiswho = pd.read_csv("cross-verified-database.csv", encoding='utf-8')

In [2]:
occupations = pd.DataFrame(whoiswho.groupby("level3_main_occ")["name"].nunique()).rename(columns={"name": "count"})
all_occ = whoiswho["level3_main_occ"].drop_duplicates().to_list()
top_occ = whoiswho.groupby("level3_main_occ")["name"].nunique().nlargest(100)
top_occ = top_occ.index.to_list()

In [3]:
occupations.sort_values(by="count",ascending=False)

count
level3_main_occ        
politician       270431
football         250439
actor            121761
writer            74709
painter           60175
...                 ...
fakultet              1
modèle                1
facteur               1
expédition            1
überwachung           1

[4985 rows x 1 columns]

There are almost 5,000 unique occupations listed as the main occupation, the most common being politician, football (…player, …coach etc.) and actor. 

In [4]:
pd.DataFrame(whoiswho[whoiswho.level3_main_occ.isin(top_occ)].groupby("level3_main_occ")["birth"].nunique().sort_values(ascending=False))

birth
level3_main_occ          
politician           1545
priest               1254
poet                 1145
aristocrat           1025
king                  962
...                   ...
judo                  116
presenter             111
canoe                 108
handball              100
computer_scientist     97

[100 rows x 1 columns]

Of the 100 most frequently listed occupation, politicians have the longest time span, with a notable politician being born in 1545 disctinct years. On the tail: computer scientists and sportspeople.

## Human Swiss army knives

Let's find the people with the highest number of listed occupations.

In [5]:
def nicerOccupations(x):
    x = x.replace("D:_","")
    x = x.split("P:")[0].split("_")
    for y in x:
        if len(y) < 2:
            x.remove(y)
    return x

In [6]:
whoiswho["all_occ"] = whoiswho["level3_all_occ"].apply(nicerOccupations)

In [7]:
whoiswho["no_occ"] = whoiswho["all_occ"].apply(lambda x: len(x))

In [8]:
whoiswho[["name","no_occ","all_occ"]].sort_values(by="no_occ",ascending=False).head(20)

name  no_occ  \
1336243  Johann_Wolfgang_von_Goethe      30   
21862                B._R._Ambedkar      26   
598608              Nikolai_Fomenko      24   
369289            Andrey_Makarevich      23   
12936         Madonna_(entertainer)      21   
485               Benjamin_Franklin      21   
7302         Alexander_von_Humboldt      20   
2244795            Alexandra_Jacobi      19   
8463                     Boris_Vian      19   
143827                     Shen_Kuo      19   
551361                 Donald_Trump      19   
964                    Dolly_Parton      18   
1854452                  Misako_Uno      18   
255903                Russell_Brand      18   
7153              Justin_Timberlake      17   
1802785              Gennady_Vetrov      17   
32293                     Al-Biruni      17   
13295            Christina_Aguilera      17   
34344                  Maya_Angelou      17   
1417258                 Nicki_Minaj      17   

                                                   all_occ  
1336243  [poet, theatre, botanist, politician, painter,...  
21862    [economist, politician, essayist, lawyer, juri...  
598608   [singer, comedian, music, presenter, radio, sk...  
369289   [singer, poet, composer, presenter, actor, arc...  
12936    [singer, actor, film, composer, film, writer, ...  
485      [writer, politician, publisher, printer, theor...  
7302     [geologist, explorer, botanist, geographer, ge...  
2244795  [man, of, letters, correspondent, journalist, ...  
8463     [polymath, singer, poet, trumpet, lyricist, pa...  
143827   [physicist, cartographer, mathematician, astro...  
551361   [magnate, investor, restaurateur, writer, entr...  
964      [guitar, singer, actor, singer, artist, banjoi...  
1854452  [vocalist, dancer, essayist, producer, designe...  
255903   [actor, columnist, comedian, presenter, film, ...  
7153     [actor, film, music, dancer, actor, entreprene...  
1802785  [writer, satirist, comedian, comedian, singer,...  
32293    [philosopher, chemist, geographer, polymath, m...  
13295    [singer, singer, composer, dancer, music, prod...  
34344    [poet, comedian, dancer, film, producer, playw...  
1417258  [composer, actor, singer, actor, entrepreneur,...

No surprise here.

## No (wo)men allowed

There are two ways to identify the professions with the largest gender gap. First we can simply take a look at the ratio of both genders. Second, we can list the most frequent occupations performed by 0 notable women or men.

In [9]:
occ_total_men = pd.Series(whoiswho[whoiswho["gender"] == "Male"].groupby(["level3_main_occ"]).size(), name="occ_total_men")
whoiswho = whoiswho.merge(occ_total_men, on = ["level3_main_occ"], how = "outer")
occ_total_women = pd.Series(whoiswho[whoiswho["gender"] == "Female"].groupby(["level3_main_occ"]).size(), name="occ_total_women")
whoiswho = whoiswho.merge(occ_total_women, on = ["level3_main_occ"], how = "outer")

In [10]:
whoiswho["gender_ratio"] = whoiswho["occ_total_men"] / whoiswho["occ_total_women"] 

In [11]:
pd.DataFrame(whoiswho.groupby("level3_main_occ")["gender_ratio"].first().nlargest(10))

gender_ratio
level3_main_occ              
prelate           3900.000000
leutnant           847.000000
organbuilder       758.000000
samurai            714.000000
landrat            711.000000
theologe           522.000000
rikishi            259.500000
gallantry          223.000000
katholischer       220.333333
bischof            218.000000

In [12]:
pd.DataFrame(whoiswho.groupby("level3_main_occ")["gender_ratio"].first().nsmallest(10))

gender_ratio
level3_main_occ              
netball              0.006135
nun                  0.006986
camogie              0.008097
actress              0.009036
beauty_pageant       0.010462
feminist             0.013025
abbess               0.013158
mistress             0.016529
wife_of              0.018587
first_lady           0.018868

In [13]:
pd.DataFrame(whoiswho[whoiswho["occ_total_men"].isnull()].groupby("level3_main_occ").size().nlargest(5))

0
level3_main_occ     
schermitrice     149
nobildonna       146
salonnière        86
suffragist        76
principessa       64

In [14]:
pd.DataFrame(whoiswho[whoiswho["occ_total_women"].isnull()].groupby("level3_main_occ").size().nlargest(5))

0
level3_main_occ      
hurler           1989
generalmajor     1323
flying_ace        905
geheimrat         428
adliger           344

In [15]:
occ_total = pd.Series(whoiswho.groupby(["level3_main_occ"]).size(), name="occ_total")
whoiswho = whoiswho.merge(occ_total, on = ["level3_main_occ"], how = "outer")

## Regionally specific occupations

In [16]:
whoiswho[whoiswho["occ_total"] > 300].groupby("level3_main_occ")["un_subregion"].nunique().nsmallest(25)

level3_main_occ
gaelic_football         1
révolution_française    2
hurler                  3
samurai                 3
kloster                 4
lacrosse                4
seiyū                   4
ämbetsman               4
earl                    5
geheimrat               5
graf                    5
nascar                  5
organbuilder            5
résistant               5
abt                     6
landrat                 6
nobile                  6
troubadour              6
adliger                 7
nobre                   7
romanist                7
feudatory               8
fiddler                 8
germanist               8
goldsmith               8
Name: un_subregion, dtype: int64

The [Gaelic football](https://en.wikipedia.org/wiki/Gaelic_football) is mentioned in the data of more than 300 notable individuals, all coming from a single subregion.

## My profession is X. Give me some side job ideas!

In [17]:
def side_occ(occupation, frame):
    side_occ_raw = []
    side_occ = {}
    whoiswho_filtered = frame[frame["level3_main_occ"] == occupation]
    combinations = whoiswho_filtered["all_occ"].to_list()
    for x in combinations:
        for y in x:
            if y != occupation:
                side_occ_raw.append(y)
    for x in side_occ_raw:
        count = side_occ_raw.count(x)
        side_occ[count] = x
    side_occ = pd.DataFrame.from_dict(side_occ, orient="index", columns=["second_occupation"]).sort_index(ascending=False)
    return(side_occ)

In [25]:
side_occ("architect", whoiswho).head(60)

second_occupation
673          academic
475          engineer
469           planner
324           painter
237          designer
160        politician
153          sculptor
124            writer
84          historian
73                art
69             artist
64         journalist
59           restorer
51          professor
26       photographer
24          education
23            teacher
22       entrepreneur
21      draughtswoman
18            militar
17             author
16           gardener
14       scenographer
13             editor
11               poet
10             critic
9             curator
8           publisher
7            surveyor
6          production
5          stonemason
4             fighter
3           physician
2               rower
1            feminist

## Relations between specific fields and professions

In [19]:
level2occs = whoiswho["level2_main_occ"].drop_duplicates().to_list()
level2occs.remove("Missing")
level2occs = sorted(level2occs)
level2_main_occ = {}
for occupation in level2occs:
    level2filtered = whoiswho[whoiswho["level2_main_occ"] == occupation]
    total = level2filtered.shape[0]
    row = {}
    
    for secondoccupation in level2occs:
        level2doublefiltered = level2filtered[level2filtered["level2_second_occ"] == secondoccupation]
        partial = level2doublefiltered.shape[0]
        ratio = partial / total
        row[secondoccupation] = ratio
    
    level2_main_occ[occupation] = row
level2_main_occ = pd.DataFrame(level2_main_occ).sort_index()
level2_main_occ.style.background_gradient(axis=None, cmap='YlOrRd')

(Vertical axis: main occupation, horizontal axis: second occupation.)

In [20]:
def correlation(frame, size):
    top = frame.groupby("level3_main_occ")["name"].nunique().nlargest(size)
    top = top.index.to_list()
    top = sorted(top)
    matrix = {}
    
    for occupation in top:
    
        whoiswho_filtered = frame[frame["level3_main_occ"] == occupation]
        total = whoiswho_filtered.shape[0]
        combinations = whoiswho_filtered["all_occ"].to_list()
    
        side_occ_raw = []
    
        for x in combinations:
            for y in x:
                if y in top:
                    side_occ_raw.append(y)
    
        row = {}
        side_occ_raw = sorted(side_occ_raw)
        
        notanymore = []
        
        for x in side_occ_raw:
            if x not in notanymore:
                count = side_occ_raw.count(x)
                ratio = count / total
                row[x] = ratio
                notanymore.append(x)
    
        matrix[occupation] = row
    
    matrix = pd.DataFrame(matrix).sort_index()
    
    return(matrix.style.background_gradient(axis=None, cmap='YlOrRd'))

In [21]:
correlation(whoiswho[(whoiswho["level2_main_occ"] != "Sports/Games")], 11)